### PR TITLE
Alessio add vimeo support

### LINKED ISSUE #358: Add support to vimeo
## Feature request
Add support to vimeo for generate transcription
### Feature description
Like youtube extend vimeo embed gutenberg block for generate the video trascription, this time the subtitles will be retrieved from the backend and sent to the plugin

### How the feature is useful
Add vimeo video transcription!

### PR DESCRIPTION
**Pull request by cpalessio96**
### All Submissions:

*   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
*   [x] Does your code has proper inline documentation? 

### Changes proposed in this Pull Request:
add support to vimeo for get vimeo video transription on gutenberg pages

Closes #358  .

### CODE REVIEW

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ includes/Api/ApiGetTranscript.php  3
         

Coverage decreased per file
===========================
- includes/Core/Sources/YouTube.php  -22
         

Complexity increasing per file
==============================
- tests/test-vimeo.php  1
- includes/Core/Sources/Vimeo.php  5
- src/block/EmbedControlPanel.js  12
- includes/Utils/VimeoHelper.php  1
- includes/Api/ApiGetTranscript.php  2
- src/block/EmbedBlock.js  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=6119951&bid=20025434)

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ includes/Api/ApiGetTranscript.php  3
         

Coverage decreased per file
===========================
- includes/Core/Sources/YouTube.php  -4
         

Complexity increasing per file
==============================
- includes/Api/ApiGetTranscript.php  2
- includes/Core/Sources/Vimeo.php  5
- src/block/EmbedBlock.js  3
- tests/test-vimeo.php  1
- includes/Utils/VimeoHelper.php  1
- src/block/EmbedControlPanel.js  12
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=6119951&bid=20025434)

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- tests/test-vimeo.php  1
- includes/Api/ApiGetTranscript.php  2
- src/block/EmbedControlPanel.js  12
- includes/Core/Sources/Vimeo.php  5
- src/block/EmbedBlock.js  3
- includes/Utils/VimeoHelper.php  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=6119951&bid=20025434)

File: includes/Api/ApiGetTranscript.php [hunk 1]
```diff
@@ -52,11 +53,27 @@ public function get_transcript_yt() {
...
+			$id_video     = basename( $url_parts['path'] );
+			$data_posts   = $this->get_data_transcript( $id_video, $from );
+			if ( $data_posts ) {
+				wp_send_json_success( $data_posts );
+			}
+			$transcript = $video_source->get_transcript( $id_video, $video_title, $from, $subtitle );
+			if ( ! $transcript['success'] ) {
+				wp_send_json_error( $transcript['message'] );
+			}
+			wp_send_json_success( $transcript['data'] );
```
Comment (HyprCoder) [inline includes/Api/ApiGetTranscript.php]: Here you should use an interface to get all the data need.

File: includes/Api/ApiGetTranscript.php [hunk 2]
```diff
@@ -107,39 +112,25 @@ public function get_video_info() {
...
-		parse_str( $url_parts['query'], $query_params );
-		if ( ! array_key_exists( 'v', $query_params ) ) {
-			wp_send_json_error( __( 'Url not a valid youtube url', 'wubtitle' ) );
+		if ( 'vimeo.com' === $url_parts['host'] ) {
+			$video_source = new Vimeo();
+			$response     = $video_source->get_video_info( $url_parts );
+			if ( ! $response['success'] ) {
+				wp_send_json_error( $response['message'] );
+			}
+			wp_send_json_success( $response );
```
Comment (HyprCoder) [inline includes/Api/ApiGetTranscript.php]: same here

**Review by HyprCoder — CHANGES_REQUESTED** [review]

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- tests/test-vimeo.php  1
- includes/Utils/VimeoHelper.php  1
- src/block/EmbedBlock.js  3
- includes/Core/Sources/Vimeo.php  5
- src/block/EmbedControlPanel.js  12
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=6119951&bid=20025434)

**Comment by GiuseppeIuculano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- includes/Core/Sources/Vimeo.php  5
- tests/test-vimeo.php  1
- src/block/EmbedBlock.js  3
- includes/Utils/VimeoHelper.php  1
- src/block/EmbedControlPanel.js  12
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/CTMobi/Wubtitle-Wordpress-Plugin/pullRequest?prid=6119951&bid=20025434)

**Review by HyprCoder — APPROVED** [review]

**Pull request merged**